domain/include/cstone/domain/assignment.hpp

```diff
@@ -98,7 +98,7 @@ class GlobalAssignment
 
         gsl::span<const KeyType> oldLeaves = tree_.treeLeaves();
         std::vector<KeyType> oldBoundaries(assignment_.numRanks() + 1);
-        for (int rank = 0; rank < oldBoundaries.size() - 1; ++rank)
+        for (KeyType rank = 0; rank < oldBoundaries.size() - 1; ++rank)
```

Comment:
`KeyType` is used exclusively for space filling curve keys, like Hilbert or Morton keys.